### PR TITLE
reviewed / extended tests and source code

### PR DESCRIPTION
**Pull request by KaiSchwarz-cnic**
* removed eslint global installation from travis as it is devDependency
* reviewed and extended automated tests
* some source code cleanup based on coverage report
* upgrade mocha to 4.0.1
* upgrade semantic-release to 8.1.1

### CODE REVIEW

File: lib/helper.js [hunk 1]
```diff
@@ -29,9 +29,9 @@ module.exports.helperCreatePasswordFile = function (options, params, PasswordFil
   if (!(options.password && options.passType)) {
     return false
   }
-  if (options.passType && options.passType.trim().length === 0) {
-    return false
-  }
+  //  if (!options.passType && options.passType.trim().length === 0) { // NOTE: ' ' is falsy and thus covered by !options.passType and thus by 2nd condition too
```
Comment (Dexus) [inline lib/helper.js:32]: Password should at last empty or 4 chars long because else will openssl error that you need a password between 4 and 1023 chars...

Comment (Dexus) [reply lib/helper.js:32]: Also you may add tests with empty password. To see if it is allowed ;)

Comment (KaiSchwarz-cnic) [reply lib/helper.js:32]: well, I think that should be covered over a new issue or through #159. makes imo more sense than adding it here.

Comment (Dexus) [reply lib/helper.js:32]: Okay, if you here finish, let me know i will final review and merge
